Server/src/models/DatasetQueryModel.ts

```diff
@@ -145,28 +145,94 @@ export class DataQueryModel {
-            .where('account.id = :idRef', { idRef: id })
-            .getRawMany();
-        return idDatasetData;
+    async getUploadedDatasetIDOfUser(userEmail: string): Promise<any[]> {
+        let userID = await this.fetchAccountIdFromEmail(userEmail)
+        if (userID == false)
+            return [false, "Invalid user email provided"]
+        else {
+            let idDatasetData: IDatasetIDModel[] = await selectDatasetIdsQuery(this.connection)
+                .innerJoin(Accounts, 'account', 'dataset.uploaderId = account.id')
+                .where('account.id = :idRef', { idRef: userID })
+                .getRawMany();
+            return [true, idDatasetData];
+        }
     }
 
     /**
-     * This method will run a query find all the data set IDs favorited by specific user ID
+     * This method will run a query find all the data set IDs favorited by a specific user
      * and return a raw data packet containing that information. 
      * 
      * @param id 
      * Account ID: number
      */
-    async getSavedDatasetIDOfUser(id: number): Promise<IDatasetIDModel[]> {
+    async getSavedDatasetIDOfUser(userEmail: string): Promise<any[]> {
+        let userID = await this.fetchAccountIdFromEmail(userEmail)
+        if (userID == false)
+            return [false, "Invalid user email provided"]
+        else {
+            let idDatasetData: IDatasetIDModel[] = await selectDatasetIdsQuery(this.connection)
+                .innerJoin('dataset.accounts', 'account')
+                .where('account.id = :idRef', { idRef: userID })
+                .getRawMany();
+            return [true, idDatasetData];
+        }
+    }
 
-        let idDatasetData: IDatasetIDModel[] = await selectDatasetIdsQuery(this.connection)
-            .innerJoin('dataset.accounts', 'account')
-            .where('account.id = :idRef', { idRef: id })
-            .getRawMany();
-        return idDatasetData;
+    private async fetchAccountIdFromEmail(userEmail: string) {
+        let userRawData = await selectAccountIdFromEmailQuery(this.connection, userEmail)
+        if (userRawData == undefined)
+            return false
+        else
+            return userRawData.id
+    }
+
+    /**
+     * This method accepts a user's email and a data set ID, will get the user ID associated with
+     * the email and, if the email was valid, will save the user ID and data set ID relation in the 
+     * accounts_datasets_dataset table if the relation does not already exist. If the email address
+     * was invalid, it will return a message reporting such. If the relation already exists, it
+     * will return a message reporting such.
+     * 
+     * @param userEmail 
+     * User's email: string
+     * @param datasetId 
+     * Data Set ID: number
+     */
+    async addSavedDatasetModel(userEmail: string, datasetId: number) {
+        let userID = await this.fetchAccountIdFromEmail(userEmail)

```

Comment:
this is great, i was hoping this is exaclty how the user id would be fetched so that later we can swap it out for the JWT.payload.id

Reply:
Currently being updated to use user ID